### PR TITLE
fix: cli - refactor config.json path

### PR DESCRIPTION
**Pull request by ananas-block**
- refactored the path from which we get config.json to ~/.config/light/config.json
- a custom path to the config json file can be provided with the LIGHT_PROTOCOL_CONFIG environment variable.
   - If this variable is set it is used, otherwise the default is used.
   - In case the enviroment variable exists but no file exists in the path an error is thrown.
   - the environment variable does not exist by default and can be added manually
- added random private key generation to cli config init

### CODE REVIEW

File: cli/src/utils/utils.ts [hunk 1]
```diff
@@ -207,20 +211,48 @@ export const setPayer = (key: string) => {
...
 
-export const setConfig = (config: Partial<Config>): void => {
+export function ensureDirectoryExists(dirPath: string): void {
+  if (!fs.existsSync(dirPath)) {
+    fs.mkdirSync(dirPath, { recursive: true });
+  }
+}
+
+export const setConfig = (config: Partial<Config>, filePath?: string): void => {
+  if (!filePath) filePath = process.env.HOME + CONFIG_PATH + CONFIG_FILE_NAME;
```
Comment (vadorovsky) [inline cli/src/utils/utils.ts:232]: What do you think about handling a custom environment variable first? I mean, having this priority list when looking up for config:

* A path set in `LIGHT_PROTOCOL_CONFIG` environment variable
* `~/.config/light/config.json`

The one specified in the env variable would have a precedence.

Comment (vadorovsky) [inline cli/src/utils/utils.ts:232]: What do you think about handling a custom environment variable first? I mean, having this priority list when looking up for config:

* A path set in `LIGHT_PROTOCOL_CONFIG` environment variable
* `~/.config/light/config.json`

The one specified in the env variable would have a precedence.

Comment (ananas-block) [reply cli/src/utils/utils.ts:232]: where would we save the environment variable, bashrc?

Comment (ananas-block) [reply cli/src/utils/utils.ts]: done, see pr description for description

**Pull request merged**